pkg/scheduler/plugins/elasticquota/plugin.go

```diff
@@ -0,0 +1,560 @@
+	RevokePodCycle   = time.Duration(1) * time.Second
+	SnapshotStateKey = "ElasticQuotaSnapshot"
+)
+
+type Plugin struct {
+	sync.RWMutex
+	handle      framework.Handle
+	pluginArgs  *config.ElasticQuotaArgs
+	quotaLister v1alpha1.ElasticQuotaLister
+	podLister   v1.PodLister
+	pdbLister   policylisters.PodDisruptionBudgetLister
+	nodeLister  v1.NodeLister
+	//only used in OnNodeAdd,in case Recover and normal Watch double call OnNodeAdd
+	nodeResourceMapLock sync.Mutex
+	nodeResourceMap     map[string]struct{}
+	groupQuotaManager   *core.GroupQuotaManager
+	// quotaOverUsedRevokeController's add/erase pod only in Reserve/ Unreserve
+	quotaOverUsedRevokeController *QuotaOverUsedRevokeController
+	// elasticQuotaController only used to update elastic quota crd
+	elasticQuotaController *Controller
+	// PreFilter, refresh elasticQuotaInfos' runtime/used according to groupQuotaManager, then write snapshot.
+	// Reserve/ Unreserve/ OnPodAdd/ OnPodUpdate/ OnPodDelete, update the corresponding SimpleElasticQuotaInfo's pods.
+	elasticQuotaInfos SimpleElasticQuotaInfos
+}
+
+// SnapshotState stores the snapshot of elasticQuotaInfos.
+type SnapshotState struct {
+	elasticQuotaInfos SimpleElasticQuotaInfos
+}
+
+// Clone the ElasticQuotaSnapshot state.
+func (s *SnapshotState) Clone() framework.StateData {
+	return &SnapshotState{
+		elasticQuotaInfos: s.elasticQuotaInfos.clone(),
+	}
+}
+
+var (
+	_ framework.PreFilterPlugin  = &Plugin{}
+	_ framework.PostFilterPlugin = &Plugin{}
+	_ framework.ReservePlugin    = &Plugin{}
+)
+
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	pluginArgs, ok := args.(*config.ElasticQuotaArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type GangSchedulingArgs, got %T", args)
+	}
+	if err := validation.ValidateElasticQuotaArgs(pluginArgs); err != nil {
+		return nil, err
+	}
+
+	client, ok := handle.(versioned.Interface)
+	if !ok {
+		kubeConfig := *handle.KubeConfig()
+		kubeConfig.ContentType = runtime.ContentTypeJSON
+		kubeConfig.AcceptContentTypes = runtime.ContentTypeJSON
+		client = versioned.NewForConfigOrDie(&kubeConfig)
+	}
+	scheSharedInformerFactory := externalversions.NewSharedInformerFactory(client, 0)
+	elasticQuotaInformer := scheSharedInformerFactory.Scheduling().V1alpha1().ElasticQuotas()
+
+	elasticQuota := &Plugin{
+		handle:            handle,
+		pluginArgs:        pluginArgs,
+		podLister:         handle.SharedInformerFactory().Core().V1().Pods().Lister(),
+		quotaLister:       elasticQuotaInformer.Lister(),
+		pdbLister:         getPDBLister(handle.SharedInformerFactory()),
+		nodeLister:        handle.SharedInformerFactory().Core().V1().Nodes().Lister(),
+		groupQuotaManager: core.NewGroupQuotaManager(pluginArgs.SystemQuotaGroupMax, pluginArgs.DefaultQuotaGroupMax),
+		nodeResourceMap:   make(map[string]struct{}),
+		elasticQuotaInfos: NewSimpleElasticQuotaInfos(),
+	}
+	elasticQuota.groupQuotaManager.SetScaleMinQuotaEnabled(true)
```

Comment:
elasticQuota.groupQuotaManager.SetScaleMinQuotaEnabled(true) should in NewGroupQuotaManager's self logic if you don't offer a config

Reply:
fixed